prow/cmd/deck/static/pr-script.js

```diff
@@ -249,6 +249,45 @@ function createSearchCard() {
     return searchCard;
 }
 
+/**
+ * GetFullPRContexts gathers build jobs and pr contexts. It firstly takes
+ * all pr contexts and only replaces contexts that have existing Prow Jobs. Tide
+ * context will be omitted from the list.
+ * @param builds
+ * @param contexts
+ * @returns {Array}
+ */
+function getFullPRContext(builds, contexts) {
+    const contextMap = new Map();
+    if (contexts) {
+        for (let context of contexts) {
+            if (context.Context === "tide") {
+                continue;
+            }
+            contextMap.set(context.Context, {
+                context: context.Context,
+                description: context.Description,
+                state: context.State.toLowerCase()
+        });
+      }
+    }
+    if (builds) {
```

Comment:
If we're overriding something here it would be awesome if we showed a warning icon and had some way for a user to discover that something was wrong... otherwise having inconsistent state is basically invisible to users and that is very frustrating

/cc @cjwagner @BenTheElder

Reply:
hmm yeah, I think if we have a github context we should prefer that and only show build statuses where either:
- there's no github context for some reason (??)
- Possibly when the github context is knowably older than the prowjob state, EG github is still showing yellow but we *know* the job finished and is passed/failed (I don't think we can trivially detect this right now, but maybe in the future). Even then the inconsistency probably makes this not worth it.

I'd also say having some way to detect the source of the state would be helpful, maybe we can mark them with the github / prow logos? 
Or in the future the avatars of the account posting the status + some other icon when it's just a prowjob state.

Reply:
@stevekuznetsov 
> a warning icon and had some way for a user to discover that something was wrong

and

> having inconsistent state is basically invisible to users

Can you explain more about those points?
@BenTheElder 
I see. For your second point, our deck couldn't update data in real-time which means that users must refresh the page to see fresh info; therefore, by the time they refresh, Github contexts may already up to date with Prow Jobs. But I still dont get why we prefer Github contextes over Prow Jobs. 

For the source of the state, I could handle that in the next PR

Reply:
well the inconsistent state that is more concerning than "it'll show up in a moment" is when we fail to post a status to github or a conflicting status is posted vs the prowjob (this can happen for various reasons, including network flakes, and that statuses are keyed by [repo, commit] which can be in more than one PR...)

Reply:
Got it. I can see it will be a very rare case that Prow Job would be used; how about displaying github contexts only?

Reply:
Well no in actual fact the ProwJob is _always_ used by `tide` -- so in any case that there is an inconsistency we should notify the user:

 - GitHub status exists but ProwJob does not
 - GitHub status does not exist but ProwJob does
 - GitHub status and ProwJob status do not agree

Reply:
I wonder if the user will be confused by knowing the incosistency. They will see the warnings, but don't have any idea why there is incositency and how to solve the problem.

Reply:
Any other input or agrement on how should we override this properly?
@BenTheElder @stevekuznetsov @cjwagner

Reply:
Tell them in the helptext to contact their `Prow` admin -- that should be their only action. But we need to make it clear that something is up

Reply:
How about this @stevekuznetsov  @BenTheElder?
![image](https://user-images.githubusercontent.com/24363693/42684949-f8521438-86d4-11e8-9765-4acef2580012.png)

Reply:
Hmmm that may end up confusing the end user more -- they may not really understand the difference between the two columns, especially if one is n/a. Is it possible to keep the current view and add a little warning icon / mouseover modal or something only if there is a discrepancy?

Reply:
Is this what you are looking for?
![image](https://user-images.githubusercontent.com/24363693/42723662-7c696494-87a5-11e8-9a3b-de3a4d49bb78.png)

Reply:
@stevekuznetsov

Reply:
bump, I think this makes sense with some hover text, can we get rid of the black behind the icon?

Reply:
Yeah, I can remove the black. Text hovering is already there but it wears off at the moment I take the screenshot

Reply:
thanks :-)
we had some problem with the 1.11.1 release images that i'm looking at now, but I'm leaving this open in a tab to review tomorrow. I want to get this in and deployed :-)